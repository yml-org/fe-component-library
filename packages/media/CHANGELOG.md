# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.1 (2023-03-13)

### Bug Fixes

- **version:** update package versions ([ebeb6fa](https://github.com/yml-org/fe-component-library/commit/ebeb6faaae123cac74215813d1d1ffaf01fee0cc))
- **version:** update package versions ([d0aeb30](https://github.com/yml-org/fe-component-library/commit/d0aeb30a06f38e6a5b5b2f309f2cd86448a848e8))
