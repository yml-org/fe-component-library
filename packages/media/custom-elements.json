{
  "schemaVersion": "1.0.0",
  "readme": "",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "media.component.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/index.ts",
      "declarations": [],
      "exports": [
        {
          "kind": "js",
          "name": "*",
          "declaration": {
            "name": "*",
            "package": "./media/media.component"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "dist/assets/index.js",
      "declarations": [
        {
          "kind": "variable",
          "name": "e"
        },
        {
          "kind": "variable",
          "name": "i"
        },
        {
          "kind": "variable",
          "name": "n"
        },
        {
          "kind": "variable",
          "name": "f",
          "default": "class extends Pt(null) {\n  constructor() {\n    super(...arguments), this.mediaTitle = \"Title\", this.mediaPartAttribute = \"webcl-media\", this.textPartAttribute = \"media-text-container\", this.titlePartAttribute = \"media-title-container\";\n  }\n  render() {\n    return $t` <div\n      class=\"inline-flex flex-row\"\n      part=${this.mediaPartAttribute}\n    >\n      <slot name=${this.imageSlotName}></slot>\n      <div part=${this.textPartAttribute}>\n        <div part=${this.titlePartAttribute}>\n          <p>\n            ${this.mediaTitle ? this.mediaTitle : this.mediaTitle === \"\" ? \"\" : Rt(Nt`Title`)}\n          </p>\n        </div>\n        <slot name=${this.staticContentSlotName}></slot>\n      </div>\n      <slot name=${this.iconSlotName}></slot>\n    </div>`;\n  }\n}"
        }
      ],
      "exports": [
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "i",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "custom-element-definition",
          "name": "e",
          "declaration": {
            "name": "n",
            "module": "dist/assets/index.js"
          }
        },
        {
          "kind": "js",
          "name": "MediaComponent",
          "declaration": {
            "name": "f",
            "module": "dist/assets/index.js"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/constants/media.variants.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "mediaProps",
          "type": {
            "text": "MediaProps"
          },
          "default": "{\n  mediaPartAttribute: 'custom-media',\n  textPartAttribute: 'custom-text',\n  titlePartAttribute: 'custom-title',\n  staticContentSlotName: 'staticContent',\n  imageSlotName: 'imageSlot',\n  iconSlotName: 'iconSlot',\n  mediaTitle: 'Media Component',\n}"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "mediaProps",
          "declaration": {
            "name": "mediaProps",
            "module": "src/constants/media.variants.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/constants/mediaIcons.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "houseIcon",
          "default": "html`<svg\r\n  width=\"80px\"\r\n  height=\"80px\"\r\n  viewBox=\"0 0 24 24\"\r\n  fill=\"none\"\r\n  xmlns=\"http://www.w3.org/2000/svg\"\r\n>\r\n  <path\r\n    d=\"M2 22H22\"\r\n    stroke=\"#292D32\"\r\n    stroke-width=\"1.5\"\r\n    stroke-miterlimit=\"10\"\r\n    stroke-linecap=\"round\"\r\n    stroke-linejoin=\"round\"\r\n  />\r\n  <path\r\n    d=\"M2.94995 22L2.99995 9.96999C2.99995 9.35999 3.28995 8.78004 3.76995 8.40004L10.77 2.95003C11.49 2.39003 12.5 2.39003 13.23 2.95003L20.23 8.39003C20.72 8.77003 21 9.34999 21 9.96999V22\"\r\n    stroke=\"#292D32\"\r\n    stroke-width=\"1.5\"\r\n    stroke-miterlimit=\"10\"\r\n    stroke-linejoin=\"round\"\r\n  />\r\n  <path\r\n    d=\"M15.5 11H8.5C7.67 11 7 11.67 7 12.5V22H17V12.5C17 11.67 16.33 11 15.5 11Z\"\r\n    stroke=\"#292D32\"\r\n    stroke-width=\"1.5\"\r\n    stroke-miterlimit=\"10\"\r\n    stroke-linecap=\"round\"\r\n    stroke-linejoin=\"round\"\r\n  />\r\n</svg>`"
        },
        {
          "kind": "variable",
          "name": "bookmarkIcon",
          "default": "html` <svg\r\n  width=\"20px\"\r\n  height=\"20px\"\r\n  viewBox=\"0 0 24 24\"\r\n  fill=\"none\"\r\n  xmlns=\"http://www.w3.org/2000/svg\"\r\n>\r\n  <path\r\n    d=\"M17 3C18.1046 3 19 3.89543 19 5L19 19.0536C19 20.5893 17.341 21.552 16.0077 20.7901L12.9923 19.067C12.3774 18.7157 11.6226 18.7157 11.0077 19.067L7.99228 20.7901C6.65897 21.552 5 20.5893 5 19.0536L5 5C5 3.89543 5.89543 3 7 3L17 3Z\"\r\n    stroke=\"#323232\"\r\n    stroke-width=\"2\"\r\n    stroke-linejoin=\"round\"\r\n  />\r\n</svg>`"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "houseIcon",
          "declaration": {
            "name": "houseIcon",
            "module": "src/constants/mediaIcons.ts"
          }
        },
        {
          "kind": "js",
          "name": "bookmarkIcon",
          "declaration": {
            "name": "bookmarkIcon",
            "module": "src/constants/mediaIcons.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/media/media.component.spec.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/media/media.component.ts",
      "declarations": [
        {
          "kind": "class",
          "description": "",
          "name": "MediaComponent",
          "members": [
            {
              "kind": "field",
              "name": "mediaTitle",
              "type": {
                "text": "string | undefined"
              },
              "default": "'Title'"
            },
            {
              "kind": "field",
              "name": "imageSlotName",
              "type": {
                "text": "string | undefined"
              }
            },
            {
              "kind": "field",
              "name": "staticContentSlotName",
              "type": {
                "text": "string | undefined"
              }
            },
            {
              "kind": "field",
              "name": "iconSlotName",
              "type": {
                "text": "string | undefined"
              }
            },
            {
              "kind": "field",
              "name": "mediaPartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'webcl-media'"
            },
            {
              "kind": "field",
              "name": "textPartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'media-text-container'"
            },
            {
              "kind": "field",
              "name": "titlePartAttribute",
              "type": {
                "text": "string | undefined"
              },
              "default": "'media-title-container'"
            },
            {
              "kind": "method",
              "name": "render"
            },
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]",
              "inheritedFrom": {
                "name": "TailwindElement",
                "module": "src/shared/tailwind.element.ts"
              }
            }
          ],
          "mixins": [
            {
              "name": "TailwindElement",
              "module": "/src/shared/tailwind.element"
            }
          ],
          "superclass": {
            "module": "src/media/media.component.ts"
          }
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "MediaComponent",
          "declaration": {
            "name": "MediaComponent",
            "module": "src/media/media.component.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/shared/tailwind.element.ts",
      "declarations": [
        {
          "kind": "mixin",
          "description": "",
          "name": "TailwindElement",
          "members": [
            {
              "kind": "field",
              "name": "styles",
              "type": {
                "text": "array"
              },
              "static": true,
              "default": "[tailwindElement, unsafeCSS(style)]"
            }
          ],
          "parameters": [
            {
              "name": "style",
              "type": {
                "text": "CSSStyleSheet | null"
              }
            }
          ]
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "TailwindElement",
          "declaration": {
            "name": "TailwindElement",
            "module": "src/shared/tailwind.element.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/types/media.ts",
      "declarations": [],
      "exports": []
    },
    {
      "kind": "javascript-module",
      "path": "src/generated/locales/es-419.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "templates",
          "type": {
            "text": "object"
          },
          "default": "{\n      's99f110d27e30b289': str`objetivo`,\n    }"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "templates",
          "declaration": {
            "name": "templates",
            "module": "src/generated/locales/es-419.ts"
          }
        }
      ]
    },
    {
      "kind": "javascript-module",
      "path": "src/generated/locales/zh-Hans.ts",
      "declarations": [
        {
          "kind": "variable",
          "name": "templates",
          "type": {
            "text": "object"
          },
          "default": "{\n      's99f110d27e30b289': str`目標`,\n    }"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "templates",
          "declaration": {
            "name": "templates",
            "module": "src/generated/locales/zh-Hans.ts"
          }
        }
      ]
    }
  ]
}
