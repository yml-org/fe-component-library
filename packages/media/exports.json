{
  ".": "./src/index.js",
  "./src/*.js": "./src/*.js",
  "./package.json": "./package.json",
  "./media.component.js": "./media.component.js"
}
