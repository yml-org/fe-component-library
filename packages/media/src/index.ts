export * from './media/media.component';
