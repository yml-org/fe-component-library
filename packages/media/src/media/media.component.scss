.parent-container {
    media-component {
        &::part(custom-media) {
            padding: 1rem;
            gap: 1.5rem;
            background-color: rgb(246, 243, 243)
        };
        &::part(custom-title) {
            font-weight: 600;
            font-size: 1.5rem;
        };
    }
    .static-slot-content {
        color: rgb(167, 166, 166);
    }
    .icon-slot-content {
        margin-top: 0.625rem;
    }
}