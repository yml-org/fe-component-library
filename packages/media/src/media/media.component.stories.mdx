import { html } from 'lit';
import { Story, Meta, Canvas } from '@storybook/addon-docs';
import './media.component.ts';
import {
  mediaProps,
  houseIcon,
  bookmarkIcon,
} from '../constants/media.variants.ts';
import './media.component.scss';

<Meta title="Components/Media" />

# Media Component

The Media component is a Lit component that renders a media component that is framework independent and utilizes Tailwind CSS internally.

# Props

| Prop Name             | Type   | Description                           |
| --------------------- | ------ | ------------------------------------- |
| mediaTitle            | string | to add title                          |
| imageSlotName         | string | the slot name for image slot          |
| staticContentSlotName | string | the slot name for static content slot |
| iconSlotName          | string | the slot name for icon slot           |
| mediaPartAttribute    | string | The part name for media component     |
| textPartAttribute     | string | The part name for text container      |
| titlePartAttribute    | string | The part name for title               |

# Examples

Default media component with title prop

export const defaultMediaTemplate = ({ mediaTitle }) =>
  html`<media-component .mediaTitle=${mediaTitle} />`;

<Canvas>
  <Story name="title" args={mediaProps}>
    {defaultMediaTemplate.bind({})}
  </Story>
</Canvas>

Media component with image slot

export const imageSlotTemplate = ({ mediaTitle, imageSlotName }) =>
  html`<media-component
    .mediaTitle=${mediaTitle}
    .imageSlotName=${imageSlotName}
  >
    <div slot=${imageSlotName}>${houseIcon}</div>
  </media-component>`;

<Canvas>
  <Story name="image slot" args={mediaProps}>
    {imageSlotTemplate.bind({})}
  </Story>
</Canvas>

Media component with static content slot

export const staticSlotTemplate = ({
  mediaTitle,
  staticContentSlotName,
  imageSlotName,
}) =>
  html`<media-component
    .mediaTitle=${mediaTitle}
    .imageSlotName=${imageSlotName}
    .staticContentSlotName=${staticContentSlotName}
  >
    <div slot=${imageSlotName}>${houseIcon}</div>
    <div slot=${staticContentSlotName}>
      <p>Emotion</p>
      <p>Time</p>
    </div>
  </media-component>`;

<Canvas>
  <Story name="static content slot" args={mediaProps}>
    {staticSlotTemplate.bind({})}
  </Story>
</Canvas>

Media component with icon slot

export const iconSlotTemplate = ({
  mediaTitle,
  staticContentSlotName,
  imageSlotName,
  iconSlotName,
}) =>
  html`<media-component
    .mediaTitle=${mediaTitle}
    .imageSlotName=${imageSlotName}
    .staticContentSlotName=${staticContentSlotName}
    .iconSlotName=${iconSlotName}
  >
    <div slot=${imageSlotName}>${houseIcon}</div>
    <div slot=${staticContentSlotName}>
      <p>Emotion</p>
      <p>Time</p>
    </div>
    <div slot=${iconSlotName}>${bookmarkIcon}</div>
  </media-component>`;

<Canvas>
  <Story name="icon slot" args={mediaProps}>
    {iconSlotTemplate.bind({})}
  </Story>
</Canvas>

Customized Media Component

export const customizedTemplate = ({
  mediaTitle,
  staticContentSlotName,
  imageSlotName,
  iconSlotName,
  mediaPartAttribute,
  textPartAttribute,
  titlePartAttribute,
}) =>
  html` <div class="parent-container">
    <media-component
      .mediaTitle=${mediaTitle}
      .imageSlotName=${imageSlotName}
      .staticContentSlotName=${staticContentSlotName}
      .iconSlotName=${iconSlotName}
      .mediaPartAttribute=${mediaPartAttribute}
      .textPartAttribute=${textPartAttribute}
      .titlePartAttribute=${titlePartAttribute}
    >
      <div slot=${imageSlotName}>${houseIcon}</div>
      <div class="static-slot-content" slot=${staticContentSlotName}>
        <p>Emotion</p>
        <p>Time</p>
      </div>
      <div class="icon-slot-content" slot=${iconSlotName}>${bookmarkIcon}</div>
    </media-component>
  </div>`;

<Canvas>
  <Story name="customized media component" args={mediaProps}>
    {customizedTemplate.bind({})}
  </Story>
</Canvas>
