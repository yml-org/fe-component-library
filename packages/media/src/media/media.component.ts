import { html } from 'lit';
import { customElement, property } from 'lit/decorators.js';
import { TailwindElement } from '../shared/tailwind.element';
import { msg, str } from '@lit/localize';

@customElement('media-component')
export class MediaComponent extends TailwindElement(null) {
  @property()
  mediaTitle?: string = 'Title';

  @property()
  imageSlotName?: string;

  @property()
  staticContentSlotName?: string;

  @property()
  iconSlotName?: string;

  @property()
  mediaPartAttribute?: string = 'webcl-media';

  @property()
  textPartAttribute?: string = 'media-text-container';

  @property()
  titlePartAttribute?: string = 'media-title-container';

  render() {
    return html` <div
      class="inline-flex flex-row"
      part=${this.mediaPartAttribute}
    >
      <slot name=${this.imageSlotName}></slot>
      <div part=${this.textPartAttribute}>
        <div part=${this.titlePartAttribute}>
          <p>${this.mediaTitle ? this.mediaTitle : this.mediaTitle === '' ? '' : msg(str`Title`)}</p>
        </div>
        <slot name=${this.staticContentSlotName}></slot>
      </div>
      <slot name=${this.iconSlotName}></slot>
    </div>`;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'media-component': MediaComponent;
  }
}
