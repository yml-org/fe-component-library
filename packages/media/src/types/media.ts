export type MediaProps = {
  mediaPartAttribute?: string;
  textPartAttribute?: string;
  titlePartAttribute?: string;
  staticContentSlotName?: string;
  imageSlotName?: string;
  iconSlotName?: string;
  mediaTitle?: string ;
};
