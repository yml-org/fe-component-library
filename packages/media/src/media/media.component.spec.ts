import { LitElement } from 'lit-element';
import { mediaProps } from '../constants/media.variants';

describe('media-component', () => {
  const MEDIA_COMPONENT = 'media-component';
  let mediaElement: LitElement;

  const getShadowRoot = (tagName: string): ShadowRoot =>
    document.body.getElementsByTagName(tagName)[0].shadowRoot;

  beforeEach(() => {
    mediaElement = window.document.createElement(MEDIA_COMPONENT) as LitElement;
    document.body.appendChild(mediaElement);
  });

  afterEach(() => {
    document.body.getElementsByTagName(MEDIA_COMPONENT)[0].remove();
  });

  it('matches component snapshot', async () => {
    expect(
      document.body.getElementsByTagName(MEDIA_COMPONENT)[0]
    ).toMatchSnapshot();
  });

  it('sets the title property correctly', async () => {
    mediaElement['mediaTitle'] = mediaProps.mediaTitle;
    await mediaElement.updateComplete;
    const elem = getShadowRoot(MEDIA_COMPONENT)
      .querySelector('div')
      .querySelector('div')
      .querySelector('div')
      .querySelector('p').innerText;
    expect(elem).toBe(mediaProps.mediaTitle);
  });

  it('sets the mediaPartAttribute property correctly', async () => {
    mediaElement['mediaPartAttribute'] = mediaProps.mediaPartAttribute;
    await mediaElement.updateComplete;
    const elem = getShadowRoot(MEDIA_COMPONENT).querySelector('div');
    expect(elem.getAttribute('part')).toBe(mediaProps.mediaPartAttribute);
  });

  it('sets the textPartAttribute property correctly', async () => {
    mediaElement['textPartAttribute'] = mediaProps.textPartAttribute;
    await mediaElement.updateComplete;
    const elem = getShadowRoot(MEDIA_COMPONENT)
      .querySelector('div')
      .querySelector('div');
    expect(elem.getAttribute('part')).toBe(mediaProps.textPartAttribute);
  });

  it('sets the titlePartAttribute property correctly', async () => {
    mediaElement['titlePartAttribute'] = mediaProps.titlePartAttribute;
    await mediaElement.updateComplete;
    const elem = getShadowRoot(MEDIA_COMPONENT)
      .querySelector('div')
      .querySelector('div')
      .querySelector('div');
    expect(elem.getAttribute('part')).toBe(mediaProps.titlePartAttribute);
  });
  
  it('renders the slot elements correctly', async () => {
    getShadowRoot(MEDIA_COMPONENT)
      .querySelector('div')
      .querySelector('slot').innerHTML =
      '<h2 class="slot-test">The Media Component</h2>';
    await mediaElement.updateComplete;
    const elem = getShadowRoot(MEDIA_COMPONENT)
      .querySelector('div')
      .querySelector('slot')
      .querySelector('h2');
    expect(elem.classList.contains('slot-test')).toBe(true);
  });
});
