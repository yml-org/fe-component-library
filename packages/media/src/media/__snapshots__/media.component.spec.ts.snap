// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media-component matches component snapshot 1`] = `
v {
  "_$AL": Map {},
  "_$Do": T {
    "_$AA": <!---->,
    "_$AB": null,
    "_$AH": gt {
      "_$AD": N {
        "el": <template />,
        "parts": Array [
          Object {
            "ctor": [Function],
            "index": 0,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 1,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 2,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 3,
            "name": "part",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "index": 5,
            "type": 2,
          },
          Object {
            "ctor": [Function],
            "index": 6,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
          Object {
            "ctor": [Function],
            "index": 7,
            "name": "name",
            "strings": Array [
              "",
              "",
            ],
            "type": 1,
          },
        ],
      },
      "_$AM": [Circular],
      "_$AN": undefined,
      "u": Array [
        R {
          "_$AH": "webcl-media",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            class="inline-flex flex-row"
            part="webcl-media"
          >
            
      
            <slot
              name=""
            />
            
      
            <div
              part="media-text-container"
            >
              
        
              <div
                part="media-title-container"
              >
                
          
                <p>
                  <!--?lit$695546061$-->
                  Title
                </p>
                
        
              </div>
              
        
              <slot
                name=""
              />
              
      
            </div>
            
      
            <slot
              name=""
            />
            
    
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "media-text-container",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            part="media-text-container"
          >
            
        
            <div
              part="media-title-container"
            >
              
          
              <p>
                <!--?lit$695546061$-->
                Title
              </p>
              
        
            </div>
            
        
            <slot
              name=""
            />
            
      
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": "media-title-container",
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <div
            part="media-title-container"
          >
            
          
            <p>
              <!--?lit$695546061$-->
              Title
            </p>
            
        
          </div>,
          "name": "part",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        T {
          "_$AA": <!--?lit$695546061$-->,
          "_$AB": null,
          "_$AH": "Title",
          "_$AM": [Circular],
          "_$AN": undefined,
          "_$Cm": true,
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 2,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
        R {
          "_$AH": undefined,
          "_$AM": [Circular],
          "_$AN": undefined,
          "element": <slot
            name=""
          />,
          "name": "name",
          "options": Object {
            "host": [Circular],
            "isConnected": true,
            "renderBefore": null,
          },
          "type": 1,
        },
      ],
    },
    "_$AM": undefined,
    "_$AN": undefined,
    "_$Cm": true,
    "options": Object {
      "host": [Circular],
      "isConnected": true,
      "renderBefore": null,
    },
    "type": 2,
  },
  "_$E_": Promise {},
  "_$Ei": undefined,
  "_$El": null,
  "__mediaPartAttribute": "webcl-media",
  "__mediaTitle": "Title",
  "__textPartAttribute": "media-text-container",
  "__titlePartAttribute": "media-title-container",
  "enableUpdating": [Function],
  "hasUpdated": true,
  "isUpdatePending": false,
  "renderOptions": Object {
    "host": [Circular],
    "isConnected": true,
    "renderBefore": null,
  },
  "renderRoot": ShadowRoot {
    "_$litPart$": T {
      "_$AA": <!---->,
      "_$AB": null,
      "_$AH": gt {
        "_$AD": N {
          "el": <template />,
          "parts": Array [
            Object {
              "ctor": [Function],
              "index": 0,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 1,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 2,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 3,
              "name": "part",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "index": 5,
              "type": 2,
            },
            Object {
              "ctor": [Function],
              "index": 6,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
            Object {
              "ctor": [Function],
              "index": 7,
              "name": "name",
              "strings": Array [
                "",
                "",
              ],
              "type": 1,
            },
          ],
        },
        "_$AM": [Circular],
        "_$AN": undefined,
        "u": Array [
          R {
            "_$AH": "webcl-media",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              class="inline-flex flex-row"
              part="webcl-media"
            >
              
      
              <slot
                name=""
              />
              
      
              <div
                part="media-text-container"
              >
                
        
                <div
                  part="media-title-container"
                >
                  
          
                  <p>
                    <!--?lit$695546061$-->
                    Title
                  </p>
                  
        
                </div>
                
        
                <slot
                  name=""
                />
                
      
              </div>
              
      
              <slot
                name=""
              />
              
    
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "media-text-container",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              part="media-text-container"
            >
              
        
              <div
                part="media-title-container"
              >
                
          
                <p>
                  <!--?lit$695546061$-->
                  Title
                </p>
                
        
              </div>
              
        
              <slot
                name=""
              />
              
      
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": "media-title-container",
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <div
              part="media-title-container"
            >
              
          
              <p>
                <!--?lit$695546061$-->
                Title
              </p>
              
        
            </div>,
            "name": "part",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          T {
            "_$AA": <!--?lit$695546061$-->,
            "_$AB": null,
            "_$AH": "Title",
            "_$AM": [Circular],
            "_$AN": undefined,
            "_$Cm": true,
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 2,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
          R {
            "_$AH": undefined,
            "_$AM": [Circular],
            "_$AN": undefined,
            "element": <slot
              name=""
            />,
            "name": "name",
            "options": Object {
              "host": [Circular],
              "isConnected": true,
              "renderBefore": null,
            },
            "type": 1,
          },
        ],
      },
      "_$AM": undefined,
      "_$AN": undefined,
      "_$Cm": true,
      "options": Object {
        "host": [Circular],
        "isConnected": true,
        "renderBefore": null,
      },
      "type": 2,
    },
  },
}
`;
